### ISSUE TITLE
Error not thrown if return value of Fn::If statement contains a syntax error

### ISSUE THREAD

**Issue by jeffreycoe** [aws-cloudformation/cfn-lint#2288]
*cfn-lint version: 0.61.0*

*Description of issue.*

If Fn::If is used within an IAM policy, and the return value from Fn::If would create a syntax issue, cfn-lint does not throw any errors indicating the final result of the IAM policy contains a syntax error.

In the example template below, the IAM policy contains the following condition block:
```yaml
- !If
  - IsDev
  - - ec2:*
    - cloudformation:*
  - cloudformation:*
```
If the IsDev condition is true, the final result will be as shown below:
```yaml
- - ec2:*
  - cloudformation:*
```
This will throw a syntax error in the CloudFormation console as it is not a valid IAM policy.

Example Template with Syntax Error:
```yaml
---
AWSTemplateFormatVersion: 2010-09-09
Description: Test Role
Parameters:
  EnvironmentId:
    Type: String
    AllowedValues:
      - DEV
      - PROD
Conditions:
  IsDev: !Equals [ !Ref EnvironmentId, "DEV" ]
Resources:
  TestRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Action: sts:AssumeRole
            Principal:
              AWS: !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:role/Test
      Policies:
        - PolicyName: TestPolicy
          PolicyDocument:
            Version: 2012-10-17
            Statement:
              - Effect: Allow
                Action:
                  - !If
                    - IsDev
                    - - ec2:*
                      - cloudformation:*
                    - cloudformation:*
                Resource: "*"
      RoleName: TestRole
```

**Comment by kddejong** [comment]
This will be validated in v1.x going forward.
